lib/api/v3/work_packages/schema/typed_work_package_schema.rb

```diff
@@ -40,6 +40,12 @@ def initialize(project:, type:)
             @type = type
           end
 
+          def writable?(property)
+            return true if property.to_sym == :date && milestone?
+
+            super
```

Comment:
Fix is looking good per se but it would probably make sense to place all this in the base contract to avoid the two schema types becoming to different. 

SpecificWorkPackageSchema currently:
``` 
return true if property == :date && work_package.leaf?
```
TypedWorkPackageSchema with the fix:
```
return true if property.to_sym == :date && milestone?
```

This could be merged to 
```
return writable?(:due_date) if property.to_sym == :date && milestone?
```

in the `writable?` method of `BaseWorkPackageSchema`